masonry/src/doc/01_creating_app.md

```diff
@@ -0,0 +1,238 @@
+# Building a "To-Do List" app
+
+**TODO - Add screenshots - see [#501](https://github.com/linebender/xilem/issues/501)**
+
+This tutorial explains how to build a simple Masonry app, step by step.
+Though it isn't representative of how we expect Masonry to be used, it does cover the basic architecture.
+
+The app we'll create is identical to the to-do-list example shown in the README.
+
+## The Widget tree
+
+Let's start with the `main()` function.
+
+```ignore
```

Comment:
```suggestion
```rust,ignore
```

I'm not sure if the Github markdown renderer (or others) handles this better (i.e. syntax highlighting), but at least it has a chance to know that this is rust code.

(Same for the other `ignore` here)

I also wonder, if we should design these inline examples such that they compile, so that they're always in sync and checked via CI.
But I see this may increase the (not so interesting) code in these examples.
That could *probably* be avoided by `# code that should not be rendered` but that on the other hand has the disadvantage, to be shown in other markdown renderers (though I'd rather accept that TBH and be sure that everything is in sync).

Reply:
> I also wonder, if we should design these inline examples such that they compile, so that they're always in sync and checked via CI.

You're welcome to try. It's *tough* to do it in a way that doesn't make the examples tedious and unreadable. (Unless you use a lot of `#` marker to hide code from the reader.)

Reply:
This also interacts with #501. That is, we definitely should make these proper examples, but that work can reasonably be deferred.

Yes, that would involve using hidden harness code.